src/lib/azure-devops/azure-devops-contributors.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  AzureDevopsTarget,
+  Username,
+  Contributor,
+  ContributorMap,
+} from "../types";
+import { Commits, Repo } from "./types";
+import { getRepoCommits, getReposPerProjects } from "./utils"; 
+
+import * as debugLib from "debug";
+const azureDefaultUrl = "https://dev.azure.com/";
+const debug = debugLib("snyk:azure-devops-count");
+
+export const fetchAzureDevopsContributors = async (
+  azureInfo: AzureDevopsTarget,
+  SnykMonitoredRepos: string[],
+  threeMonthsDate: string
+): Promise<ContributorMap> => {
+  const contributorsMap = new Map<Username, Contributor>();
+  try {
+    let repoList: Repo[] = [];
+
+    if (
+      azureInfo.repo &&
+      (!azureInfo.projectKeys ||
+        azureInfo.projectKeys.length > 1)
+    ) {
+      // If repo is specified, then a single project key is expected, bail otherwise
+      console.log("You must provide a single project for single repo counting");
+      process.exit(1);
+    } else if (azureInfo.repo) {
+      // If repo is specified, and we got a single project key
+      debug("Counting contributors for single repo");
+      if(azureInfo.projectKeys){
+      repoList.push({
+        name: azureInfo.repo,
+        project: { key: azureInfo.projectKeys[0] },
+      });
+    }
+    else{
+      debug("Oops, seems we couldn't find a project name. Please retry");
+    }
+    } else {
```

Comment:
these else are confusing. It this indented with prettier?

Reply:
I thought about a scenario where the repo is stated but the project is not and wanted to alert the user.
Can remove it if you want.

Reply:
that's fine, but the line identations are not looking right. The first else should be a bit more to the right, aligned with the if above

Reply:
done